In [ ]:
import os
import pandas as pd

pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

**Dataset completo overview**   
 Quais variáreis estão preenchidas e possuem valores distintos? Quantos modelos? Tem um modelo com maior taxa de erro? Quantos discos? Todos os discos tem a mesma temporalidade de registros? Quando tem erro, o erro permanece por quanto tempo? 

In [ ]:
# carrega dados de setembro
path_raw_data = '../data/data_q3_2024'
csv_files = [arquivo for arquivo in os.listdir(path_raw_data) if arquivo.startswith("2024-09-")]
df = pd.DataFrame()
for file in csv_files:
    print(file)
    df_aux = pd.read_csv(f'{path_raw_data}/{file}')
    df = pd.concat([df, df_aux])

print(df.shape)
df.head()

2024-09-01.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-02.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-03.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-04.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-05.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-06.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-07.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-08.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-09.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-10.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-11.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-12.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-13.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-14.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-15.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-16.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-17.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-18.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-19.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-20.csv
2024-09-21.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-22.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-23.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-24.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-25.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-26.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-27.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-28.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-29.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


2024-09-30.csv


C:\Users\USER\AppData\Local\Temp\ipykernel_18128\3524516862.py:5: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aux = pd.read_csv(f'../data/data_q3_2024/{file}')


(8780281, 197)


date   serial_number                 model  capacity_bytes  failure  \
0  2024-09-01  PL1331LAGXN3AH  HGST HDS5C4040ALE630   4000787030016        0   
1  2024-09-01  PL2331LAGSTSYJ  HGST HMS5C4040ALE640   4000787030016        0   
2  2024-09-01  S2ZYJ9FG405079         ST500LM012 HN    500107862016        0   
3  2024-09-01        Z3056CGY           ST4000DM000   4000787030016        0   
4  2024-09-01        Z305FT10           ST4000DM000   4000787030016        0   

  datacenter  cluster_id  vault_id  pod_id  pod_slot_num  is_legacy_format  \
0       sac0           0      1021       0          39.0             False   
1       sac0           0      1021       0           0.0             False   
2       sac0           0      1021       0           NaN             False   
3       sac0           0      1021       0          21.0             False   
4       sac0           0      1021       0          29.0             False   

   smart_1_normalized  smart_1_raw  smart_2_normalized  smart_2_raw  \
0               100.0          0.0               133.0        105.0   
1               100.0          0.0               134.0        102.0   
2               100.0         68.0               252.0          0.0   
3               120.0  242183320.0                 NaN          NaN   
4               116.0  112501984.0                 NaN          NaN   

   smart_3_normalized  smart_3_raw  smart_4_normalized  smart_4_raw  \
0               130.0        535.0               100.0         15.0   
1               128.0        541.0               100.0         25.0   
2                94.0       2020.0               100.0         20.0   
3                92.0          0.0               100.0         17.0   
4                91.0          0.0               100.0         17.0   

   smart_5_normalized  smart_5_raw  smart_7_normalized  smart_7_raw  \
0               100.0          0.0               100.0          0.0   
1               100.0          0.0               100.0          0.0   
2               252.0          0.0               252.0          0.0   
3               100.0          0.0                83.0  226609511.0   
4               100.0          0.0                84.0  302836068.0   

   smart_8_normalized  smart_8_raw  smart_9_normalized  smart_9_raw  \
0               113.0         42.0                91.0      64875.0   
1               113.0         42.0                88.0      84568.0   
2               252.0          0.0               100.0      43713.0   
3                 NaN          NaN                15.0      74698.0   
4                 NaN          NaN                15.0      74700.0   

   smart_10_normalized  smart_10_raw  smart_11_normalized  smart_11_raw  \
0                100.0           0.0                  NaN           NaN   
1                100.0           0.0                  NaN           NaN   
2                252.0           0.0                 98.0        2899.0   
3                100.0           0.0                  NaN           NaN   
4                100.0           0.0                  NaN           NaN   

   smart_12_normalized  smart_12_raw  smart_13_normalized  smart_13_raw  \
0                100.0          15.0                  NaN           NaN   
1                100.0          25.0                  NaN           NaN   
2                100.0          21.0                  NaN           NaN   
3                100.0          17.0                  NaN           NaN   
4                100.0          17.0                  NaN           NaN   

   smart_15_normalized  smart_15_raw  smart_16_normalized  smart_16_raw  \
0                  NaN           NaN                  NaN           NaN   
1                  NaN           NaN                  NaN           NaN   
2                  NaN           NaN                  NaN           NaN   
3                  NaN           NaN                  NaN           NaN   
4                  NaN           NaN                  NaN           NaN   

___
Características gerais

In [ ]:
# lista variáveis no csv original e principais informações

def summarize_dataframe(df):
    summary = pd.DataFrame({
        'Column': df.columns,
        'Distinct Values': [df[col].nunique() for col in df.columns],
        'Null Values': [df[col].isnull().sum() for col in df.columns],
        '% Values': [df[col].notnull().sum()/len(df) for col in df.columns],
        'Data Type': [df[col].dtype for col in df.columns]
    })
    return summary

# Resumo do DataFrame
summary = summarize_dataframe(df)
print(summary)

                   Column  Distinct Values  Null Values  % Values Data Type
0                    date               30            0  1.000000    object
1           serial_number           297061            0  1.000000    object
2                   model               78            0  1.000000    object
3          capacity_bytes               18            0  1.000000     int64
4                 failure                2            0  1.000000     int64
5              datacenter                5       797124  0.909214    object
6              cluster_id                6            0  1.000000     int64
7                vault_id              230            0  1.000000     int64
8                  pod_id               20            0  1.000000     int64
9            pod_slot_num               60       126366  0.985608   float64
10       is_legacy_format                1            0  1.000000      bool
11     smart_1_normalized               84        26550  0.996976   float64
12          

Algumas variáveis tem muitos valores nulos ou um valor constante. 
Exemplo 198 possui menos que 10% dos valores preenchidos.

In [ ]:
# resumo informções por modelo

df_model = df.groupby('model').agg({'date':['min','max'], 'serial_number':['nunique', 'count'], 'failure':'sum'}).reset_index()
df_model.columns = ['model', 'first_date', 'last_date', 'disks', 'records', 'failure_records']
df_model['failure_rate'] = df_model['failure_records']/df_model['records']*100
df_model.sort_values('failure_rate', ascending=False)

model  first_date   last_date  disks  \
77                         WUH721816ALE6L4  2024-09-01  2024-09-30     72   
12                    HGST HUH728080ALE604  2024-09-01  2024-09-30     85   
71                          WDC WD5000LPVX  2024-09-01  2024-09-30     68   
66                      TOSHIBA MQ01ABF050  2024-09-01  2024-09-30    163   
67                     TOSHIBA MQ01ABF050M  2024-09-01  2024-09-30    168   
25                           ST12000NM000J  2024-09-01  2024-09-30    632   
23                           ST12000NM0007  2024-09-01  2024-09-30   1072   
53      Seagate BarraCuda SSD ZA250CM10002  2024-09-01  2024-09-30    505   
8                     HGST HUH721212ALE600  2024-09-01  2024-09-30   2614   
32                           ST14000NM0138  2024-09-01  2024-09-30   1348   
10                    HGST HUH721212ALN604  2024-09-01  2024-09-30  10355   
11                    HGST HUH728080ALE600  2024-09-01  2024-09-30   1094   
24                           ST12000NM0008  2024-09-01  2024-09-30  19271   
9                     HGST HUH721212ALE604  2024-09-01  2024-09-30  13264   
50  Seagate BarraCuda 120 SSD ZA250CM10003  2024-09-01  2024-09-30   1095   
31                           ST14000NM001G  2024-09-01  2024-09-30  10644   
63                    TOSHIBA MG08ACA16TEY  2024-09-01  2024-09-30   5178   
61                     TOSHIBA MG08ACA16TA  2024-09-01  2024-09-30  40161   
0                           CT250MX500SSD1  2024-09-01  2024-09-30    693   
60                    TOSHIBA MG07ACA14TEY  2024-09-01  2024-09-30    730   
44                             ST8000DM002  2024-09-01  2024-09-30   9139   
62                     TOSHIBA MG08ACA16TE  2024-09-01  2024-09-30   5934   
76                     WDC WUH722222ALE6L4  2024-09-01  2024-09-30  19220   
74                     WDC WUH721816ALE6L0  2024-09-01  2024-09-30   3024   
65                     TOSHIBA MG10ACA20TE  2024-09-01  2024-09-30   3604   
26                           ST12000NM001G  2024-09-01  2024-09-30  13201   
21                           ST10000NM0086  2024-09-01  2024-09-30   1053   
47                            ST8000NM0055  2024-09-01  2024-09-30  13617   
73                     WDC WUH721414ALE6L4  2024-09-01  2024-09-30   8513   
59                     TOSHIBA MG07ACA14TA  2024-09-01  2024-09-30  37835   
34                           ST16000NM001G  2024-09-01  2024-09-30  33564   
75                     WDC WUH721816ALE6L4  2024-09-01  2024-09-30  26409   
5                     HGST HMS5C4040BLE640  2024-09-01  2024-09-30   7736   
7                     HGST HUH721010ALE600  2024-09-01  2024-09-30     20   
54      Seagate BarraCuda SSD ZA500CM10002  2024-09-01  2024-09-30     12   
55   Seagate FireCuda 120 SSD ZA500GM10001  2024-09-01  2024-09-30      6   
56           Seagate IronWolf ZA250NM10002  2024-09-01  2024-09-30    132   
57                             Seagate SSD  2024-09-01  2024-09-30    101   
58                         TOSHIBA HDWF180  2024-09-01  2024-09-30     57   
19                          SSDSCKKB240GZR  2024-09-01  2024-09-30      1   
6                     HGST HMS5C4040BLE641  2024-09-01  2024-09-30      1   
64                     TOSHIBA MG09ACA16TE  2024-09-01  2024-09-30      6   
51  Seagate BarraCuda 120 SSD ZA500CM10003  2024-09-01  2024-09-30      3   
4                     HGST HMS5C4040ALE640  2024-09-01  2024-09-30    598   
3                     HGST HDS724040ALE640  2024-09-01  2024-09-30      1   
68                 WD Blue SA510 2.5 250GB  2024-09-01  2024-09-30    126   
69                          WDC WD5000BPKT  2024-09-01  2024-09-30      7   
70                          WDC WD5000LPCX  2024-09-01  2024-09-30     51   
2                     HGST HDS5C4040ALE630  2024-09-01  2024-09-19      4   
72                         WDC WDS250G2B0A  2024-09-01  2024-09-30    177   
52     Seagate BarraCuda SSD ZA2000CM10002  2024-09-01  2024-09-30      3   
48                 Samsung SSD 85

Há uma variação no número de discos e registros por modelo. A taxa de falha também muda de acordo com o modelo.

In [ ]:
# Há discos que não aparecem nos 30 dias do mês?
df.serial_number.value_counts().reset_index().sort_values('count').head(10)

serial_number  count
297060      1RK1R6XG      1
297033      70S002L8      1
297034  6410A01SFVKG      1
297035      8CKL8H2F      1
297036      ZL285Z7T      1
297037  8120A00DFVKG      1
297038  14C0A011FVKG      1
297039      QBGU2NWT      1
297040      2BKWBSYT      1
297041      ZL2P7R2C      1

In [ ]:
# Quantos discos que não aparecem nos 30 dias do mês?
len(df.serial_number.value_counts().reset_index()[df.serial_number.value_counts().reset_index()['count']<30])

10165

In [ ]:
# Exemplo: periodicidade e indicativa de falha para discos que aparecem 25 dias 

teste_25dias = df.serial_number.value_counts().reset_index()[df.serial_number.value_counts().reset_index()['count']==25].head(10).serial_number
df[df.serial_number.isin(teste_25dias)].groupby('serial_number').agg({'date':['min','max'], 'failure':['count','sum']}).reset_index()

serial_number        date             failure    
                        min         max   count sum
0  34M0A0BYFJKH  2024-09-06  2024-09-30      25   0
1      5QKANRAB  2024-09-06  2024-09-30      25   0
2  70E0A0HBF97G  2024-09-06  2024-09-30      25   0
3      8CKN10AE  2024-09-06  2024-09-30      25   0
4      AAHM4DNH  2024-09-01  2024-09-25      25   1
5      QGK9S7WT  2024-09-01  2024-09-25      25   1
6      ZHZ39RG8  2024-09-01  2024-09-25      25   1
7      ZL2D7J9W  2024-09-01  2024-09-30      25   0
8      ZLW0GJLW  2024-09-01  2024-09-25      25   1
9      ZLW18MG9  2024-09-01  2024-09-25      25   1

As vezes os registros começam depois, as vezes param antes (nos exemplos acima, os que param antes do final do mês, falharam)

___
Explorando erros

In [ ]:
# Total de falhas  
print('total de falhas:', df['failure'].sum())

total de falhas: 351


In [ ]:
# Quantos discos falharam mais de uma vez?
len(df.groupby('serial_number')['failure'].sum().reset_index()[df.groupby('serial_number')['failure'].sum().reset_index()['failure']>1])

0

In [36]:
# o último registro é o dia da falha?

# Determina a data máxima por ID
max_date_disk = df.groupby(['model','serial_number'])['date'].max().reset_index()
max_date_disk.rename(columns={'date': 'max_date'}, inplace=True)

# Determina a data de falha por ID (caso exista falha=1)
failure_date = df[df['failure'] == 1].groupby('serial_number')['date'].max().reset_index()
failure_date.rename(columns={'date': 'failure_date'}, inplace=True)

# Junta os dois resultados no dataframe original
output = pd.merge(max_date_disk, failure_date, on='serial_number', how='right')

# Adiciona uma coluna indicando se há registros após a falha
print(len(failure_date))
print(len(output[output.max_date > output.failure_date]))
output[output.max_date > output.failure_date]

351
7


model serial_number    max_date failure_date
104  HGST HUH721212ALN604      5QKV46NB  2024-09-17   2024-09-13
172  HGST HUH721212ALE604      8HHEHEWH  2024-09-17   2024-09-13
186  HGST HUH721212ALE604      AAG3ZSKH  2024-09-24   2024-09-20
197  HGST HUH721212ALN604      AAHM4WJH  2024-09-17   2024-09-16
251         ST12000NM0008      ZHZ3N9S2  2024-09-23   2024-09-20
276         ST12000NM0008      ZJ003HW4  2024-09-14   2024-09-13
289         ST12000NM0008      ZL0051LC  2024-09-16   2024-09-15

351 falhas ao todo. Nenhum disco falha mais de uma vez. Sete casos que tem registro depois da falha (não necessariamente continuam operando até o final do mês). 

In [ ]:
# exibindo casos que falham e continuam operando

df[df.serial_number.isin(output[output.max_date > output.failure_date].serial_number)].sort_values(['serial_number','date'])

date serial_number                 model  capacity_bytes  \
78971   2024-09-01      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78937   2024-09-02      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78843   2024-09-03      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78842   2024-09-04      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78845   2024-09-05      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78846   2024-09-06      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78969   2024-09-07      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78937   2024-09-08      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78964   2024-09-09      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78964   2024-09-10      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78963   2024-09-11      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78962   2024-09-12      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78957   2024-09-13      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78031   2024-09-15      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78026   2024-09-16      5QKV46NB  HGST HUH721212ALN604  12000138625024   
78022   2024-09-17      5QKV46NB  HGST HUH721212ALN604  12000138625024   
138448  2024-09-01      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138440  2024-09-02      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138344  2024-09-03      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138404  2024-09-04      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138474  2024-09-05      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138479  2024-09-06      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138600  2024-09-07      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138505  2024-09-08      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138532  2024-09-09      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138598  2024-09-10      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138596  2024-09-11      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138594  2024-09-12      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138586  2024-09-13      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138584  2024-09-14      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
137659  2024-09-15      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
137650  2024-09-16      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
137643  2024-09-17      8HHEHEWH  HGST HUH721212ALE604  12000138625024   
138459  2024-09-01      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138451  2024-09-02      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138355  2024-09-03      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138415  2024-09-04      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138485  2024-09-05      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138490  2024-09-06      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138611  2024-09-07      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138516  2024-09-08      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138543  2024-09-09      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138609  2024-09-10      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138607  2024-09-11      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138605  2024-09-12      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138597  2024-09-13      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
138595  2024-09-14      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
137670  2024-09-15      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
137664  2024-09-18      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
137665  2024-09-19      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
134605  2024-09-20      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
134597  2024-09-21      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
134578  2024-09-22      AAG3ZSKH  HGST HUH721212ALE604  12000138625024   
134540  2024-09-23